contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,181 @@
+
+    /// @notice Emitted when a user emergency withdraws its moJOE
+    event EmergencyWithdraw(address indexed user, uint256 amount);
+
+    /**
+     * @notice Initialize a new StableJoeStaking contract.
+     * @dev This contract needs to receive an ERC20 `_rewardToken` in order to distribute them
+     * (with JoeMakerV4 in our case)
+     * @param _rewardToken The address of the ERC-20 reward token
+     * @param _moJoe The address of the ERC-20 moJOE token, traderjoe's staking token.
+     */
+    function initialize(IERC20Upgradeable _rewardToken, IERC20Upgradeable _moJoe) public initializer {
+        __Ownable_init();
+
+        rewardToken = _rewardToken;
+        moJoe = _moJoe;
+    }
+
+    /**
+     * @notice View function to see pending reward token on frontend.
+     * @param _user The address of the user
+     * @return `_user`'s pending reward token
+     */
+    function pendingToken(address _user) external view returns (uint256) {
+        UserInfo storage user = userInfo[_user];
+        uint256 lpSupply = moJoe.balanceOf(address(this));
+        uint256 _accTokenPerShare = accTokenPerShare;
+
+        uint256 rewardTokenBalance = rewardToken.balanceOf(address(this));
+        if (rewardTokenBalance != lastTokenBalance && lpSupply != 0) {
+            uint256 rewardTokenAdded = rewardTokenBalance.sub(lastTokenBalance);
+            _accTokenPerShare = _accTokenPerShare.add(rewardTokenAdded.mul(1e12).div(lpSupply));
+        }
+        return user.amount.mul(_accTokenPerShare).div(1e12).sub(user.rewardDebt);
+    }
+
+    /**
+     * @notice Update reward variables to be up-to-date.
+     * @dev Needs to be called before any deposit or withdrawal.
+     */
+    function updatePool() public {
+        uint256 rewardTokenBalance = rewardToken.balanceOf(address(this));
+        if (rewardTokenBalance == lastTokenBalance) {
+            return;
+        }
+
+        uint256 lpSupply = moJoe.balanceOf(address(this));
+        if (lpSupply == 0) {
+            lastTokenBalance = rewardTokenBalance;
+            return;
+        }
+
+        uint256 rewardTokenAdded = rewardTokenBalance.sub(lastTokenBalance);
+        accTokenPerShare = accTokenPerShare.add(rewardTokenAdded.mul(1e12).div(lpSupply));
+        lastTokenBalance = rewardTokenBalance;
+    }
+
+    /**
+     * @notice Deposit moJoe to sJoe for reward token allocation.
+     * @param _amount The amount of moJOE to deposit
+     */
+    function deposit(uint256 _amount) public {
+        UserInfo storage user = userInfo[msg.sender];
+
+        updatePool();
+        if (user.amount > 0) {
+            uint256 pending = user.amount.mul(accTokenPerShare).div(1e12).sub(user.rewardDebt);
+            safeTokenTransfer(msg.sender, pending);
+        }
+        user.amount = user.amount.add(_amount);
+        user.rewardDebt = user.amount.mul(accTokenPerShare).div(1e12);
+
+        moJoe.safeTransferFrom(address(msg.sender), address(this), _amount);
```

Comment:
nit: i don't think you need the `address` around `address(msg.sender)` since `msg.sender` should already be an address I believe